[FLINK-14395] Refactor ES 6 connector to split table-specific code into flink-sql-connector-elasticsearch6

yanghua opened the pull request:
## What is the purpose of the change

*This pull request try to refactor ES 6 connector to split table-specific code into flink-sql-connector-elasticsearch6*

## Brief change log

  - *Moved table-specific code from elasticsearch 6.x general connector to sql connector*
  - *changed pom file and some resource files*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

yanghua commented:
I have reported the Kafka test issue on Travis. Trying to rebuild now.

yanghua commented:
@flinkbot run travis

yanghua commented:
@flinkbot run travis

yanghua commented:
Still failed because of the same Kafka relevant test failure.

yanghua commented:
@flinkbot run travis

aljoscha commented on flink-connectors/flink-sql-connector-elasticsearch6/src/test/java/org/apache/flink/streaming/sql/connectors/elasticsearch6/Elasticsearch6UpsertTableSinkFactoryTest.java:
```diff
@@ -65,11 +65,11 @@ public void testBuilder() {
-			DOC_TYPE,
-			KEY_DELIMITER,
-			KEY_NULL_LITERAL,
+			Collections.singletonList(new Host(ElasticsearchUpsertTableSinkFactoryTestBase.HOSTNAME, ElasticsearchUpsertTableSinkFactoryTestBase.PORT, ElasticsearchUpsertTableSinkFactoryTestBase.SCHEMA)),
```
Shouldn't this work without the `ElasticsearchUpsertTableSinkFactoryTestBase` prefix? We have `ElasticsearchUpsertTableSinkFactoryTestBase` as our base class.

yanghua replied:
You are right. The IDE added the prefix when I moving the class files.

aljoscha commented on flink-connectors/flink-sql-connector-elasticsearch6/pom.xml:
```diff
@@ -36,11 +36,65 @@ under the License.
 			<artifactId>flink-connector-elasticsearch6_${scala.binary.version}</artifactId>
 			<version>${project.version}</version>
 		</dependency>
+
```
These should now be removed from the pure ES6 connector because they are not needed anymore.

yanghua replied:
In my memory, I used to try to remove these dependencies in the PR #9720 , but this behavior caused test case failure, so I gave up, let me try again.

yanghua commented:
> I don't think this will still work because this was not changed:
> 
> https://github.com/apache/flink/blob/10f9f1d431dbf047ea14130d9fb7fdd4f78178fc/flink-connectors/flink-connector-elasticsearch6/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory#L16
> 
> . And you changed the name of the package.

I have moved the file from `flink-connector-elasticsearch6` to `flink-sql-connector-elasticsearch6` just like what I did in the PR #9720 . So it should not be a problem?

yanghua commented:
@aljoscha Thanks for reviewing. @twalthr and Jark have different opinions about the purpose of this PR want to do. Can you have a look at our discussion in the JIRA.

aljoscha commented:
> > I don't think this will still work because this was not changed:
> > https://github.com/apache/flink/blob/10f9f1d431dbf047ea14130d9fb7fdd4f78178fc/flink-connectors/flink-connector-elasticsearch6/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory#L16
> > 
> > . And you changed the name of the package.
> 
> I have moved the file from `flink-connector-elasticsearch6` to `flink-sql-connector-elasticsearch6` just like what I did in the PR #9720 . So it should not be a problem?

Yes, but for the other PR you changed the service file: https://github.com/apache/flink/blob/e2c4213a8f0664285658f25f2169487eb952909e/flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory#L16

yanghua commented:
> > > I don't think this will still work because this was not changed:
> > > https://github.com/apache/flink/blob/10f9f1d431dbf047ea14130d9fb7fdd4f78178fc/flink-connectors/flink-connector-elasticsearch6/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory#L16
> > > 
> > > . And you changed the name of the package.
> > 
> > 
> > I have moved the file from `flink-connector-elasticsearch6` to `flink-sql-connector-elasticsearch6` just like what I did in the PR #9720 . So it should not be a problem?
> 
> Yes, but for the other PR you changed the service file:
> 
> https://github.com/apache/flink/blob/e2c4213a8f0664285658f25f2169487eb952909e/flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory#L16


> > Yes, but for the other PR you changed the service file:

In this PR, I also changed the service file, now it is:

```
org.apache.flink.streaming.sql.connectors.elasticsearch6.Elasticsearch6UpsertTableSinkFactory
```

yanghua commented on flink-connectors/flink-sql-connector-elasticsearch6/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory:
```diff
@@ -13,4 +13,4 @@
 # limitations under the License.
 
-org.apache.flink.streaming.connectors.elasticsearch6.Elasticsearch6UpsertTableSinkFactory
+org.apache.flink.streaming.sql.connectors.elasticsearch6.Elasticsearch6UpsertTableSinkFactory
```
@aljoscha see here, is anything wrong?

yanghua commented:
@flinkbot run travis

yanghua commented:
This PR go to wrong direction, closing it.
